format/avro/ocf.go

```diff
@@ -0,0 +1,117 @@
+package avro
+
+import (
+	"github.com/wader/fq/format"
+	"github.com/wader/fq/format/avro/decoders"
+	"github.com/wader/fq/format/avro/schema"
+	"github.com/wader/fq/format/registry"
+	"github.com/wader/fq/pkg/decode"
+	"github.com/wader/fq/pkg/scalar"
+)
+
+var jsonGroup decode.Group
+
+func init() {
+	registry.MustRegister(decode.Format{
+		Name:        format.AVRO_OCF,
```

Comment:
I've been thinking about format decoders that themselves are encoding some generic data. Maybe they should have some help functions in jq to convert the decoded structure into something that it represent. For example avro could have something like this in a `avro.jq` file that it embeds (see https://github.com/wader/fq/pull/37)
```jq
# maybe need some handling of nested things
# and the _len is not decided yet
def avro_ocf_torepr:
  ( .blocks
  | map(
    ( .data
    | map(with_entries(select(.key|endswith("_len") | not))
    )
  )
```
That would make is possible to do:
```sh
$ fq twitter.avro avro_torepr
[
  [
    {
      "timestamp": 1366150681,
      "tweet": "Rock: Nerf paper, scissors is fine.",
      "username": "miguno"
    },
    {
      "timestamp": 1366154481,
      "tweet": "Works as intended.  Terran is IMBA.",
      "username": "BlizzardCS"
    }
  ]
]
```

Not sure about the `*_torepr` convetion yet, `*_tovalue` or `*_tojson` i think is confusing as value and json used for other things. `tovalue` used by fq and `tojson/fromjson` in jq is to/from jsong representation.

Reply:
Maybe the function should be just `torepr` and somehow figure if the format has a repr function